[WPE] Fix build after 301229

ItsJamie9494 opened the pull request:
#### 224689d07870fdadc75b7b0fbbff9c935237daac
<pre>
[WPE] Fix build after 301229
<a href="https://bugs.webkit.org/show_bug.cgi?id=300684">https://bugs.webkit.org/show_bug.cgi?id=300684</a>

Reviewed by Michael Catanzaro.

WPE didn&apos;t have the WebExtensions code enabled. Enabling it
reveals many bugs that assumed Gtk would always be present.
We can move a lot of code to GTK specific files and leave non-GTK parts
available for WPE.

This also disables the action-parsing test on WPE, since that
test creates a PNG image and WPE doesn&apos;t have any libraries that
support doing that.

* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/graphics/Icon.cpp:
* Source/WebCore/platform/graphics/Icon.h:
* Source/WebCore/platform/graphics/glib/IconGLib.cpp: Renamed from Source/WebCore/platform/graphics/gtk/IconGtk.cpp.
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/Shared/API/APIData.h:
* Source/WebKit/Shared/Extensions/wpe/WebExtensionUtilitiesWPE.cpp:
(WebKit::availableScreenScales):
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/Extensions/glib/WebExtensionGLib.cpp: Copied from Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp.
* Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGTK.cpp: Added.
(WebKit::WebExtension::iconForPath):
* Source/WebKit/UIProcess/Extensions/wpe/WebExtensionWPE.cpp: Added.
(WebKit::WebExtension::iconForPath):
* Source/cmake/OptionsWPE.cmake:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebExtension.cpp:
(beforeAll):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/PlatformGTK.cmake:
* Tools/TestWebKitAPI/glib/WebKitGLib/gtk/WebExtensionUtilities.cpp: Renamed from Tools/TestWebKitAPI/glib/WebKitGLib/WebExtensionUtilities.cpp.

Canonical link: <a href="https://commits.webkit.org/301906@main">https://commits.webkit.org/301906@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d57b6b30d4b4638d709fdc1b93e608f48e61b734

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126880 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46520 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37502 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133884 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78474 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128751 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47143 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55051 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/96553 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64950 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129828 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37723 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113500 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77067 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36590 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31674 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77278 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/118923 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107557 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31985 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136410 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/125344 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53546 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41223 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105066 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54046 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109862 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104762 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26818 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50268 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28621 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/51021 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53476 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59341 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/158383 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52725 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39618 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56060 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54477 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebCore/SourcesGTK.txt:
```diff
@@ -80,11 +80,14 @@ platform/graphics/gbm/GraphicsContextGLTextureMapperGBM.cpp @no-unify
 platform/graphics/gtk/GdkCairoUtilities.cpp
+platform/graphics/gtk/GdkSkiaUtilities.cpp
 platform/graphics/gtk/IconGtk.cpp
+platform/graphics/gtk/ImageAdapterGtk.cpp
```
These are conflicts with [301527@main](https://commits.webkit.org/301527@main) and [301544@main](https://commits.webkit.org/301544@main). You'll need to rebase.

ItsJamie9494 replied:
Ah, dammit, I thought I had rebased it for this.

mcatanzaro commented on Source/WebKit/SourcesGTK.txt:
```diff
@@ -230,6 +230,7 @@ UIProcess/Automation/gtk/WebAutomationSessionGtk.cpp @no-unify
 UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp
 
+UIProcess/Extensions/glib/WebExtensionGLib.cpp
 UIProcess/Extensions/gtk/WebExtensionGtk.cpp
```
```suggestion
UIProcess/Extensions/glib/WebExtensionGLib.cpp

UIProcess/Extensions/gtk/WebExtensionGtk.cpp
```

Each directory is separated by a blank line.

mcatanzaro commented on Source/WebKit/SourcesWPE.txt:
```diff
@@ -223,6 +223,9 @@ UIProcess/Automation/glib/BidiBrowserAgentGlib.cpp @no-unify
 
 UIProcess/CoordinatedGraphics/DrawingAreaProxyCoordinatedGraphics.cpp
 
+UIProcess/Extensions/glib/WebExtensionGLib.cpp
+UIProcess/Extensions/wpe/WebExtensionWPE.cpp
```
```suggestion
UIProcess/Extensions/glib/WebExtensionGLib.cpp

UIProcess/Extensions/wpe/WebExtensionWPE.cpp
```

carlosgcampos commented on Source/WebKit/Shared/Extensions/wpe/WebExtensionUtilitiesWPE.cpp:
```diff
@@ -28,6 +28,7 @@
 #if ENABLE(WK_WEB_EXTENSIONS)
 
+#include "WPEDisplay.h"
 #include "WPEToplevelPrivate.h"
```
You should not add the wpe platform headers directly, there's a single header `<wpe/wpe-platform.h>` that should be included conditionally using  `#if ENABLE(WPE_PLATFORM)`

carlosgcampos commented on Source/WebKit/Shared/Extensions/wpe/WebExtensionUtilitiesWPE.cpp:
```diff
@@ -40,9 +41,7 @@ Vector<double> availableScreenScales()
 
     GUniquePtr<GList> toplevels(wpe_toplevel_list());
```
We shouldn't use wpe platform api unconditionally, we needs to check if the app is using it see `WKWPE::isUsingWPEPlatformAPI()`

carlosgcampos commented on Source/WebKit/Shared/Extensions/wpe/WebExtensionUtilitiesWPE.cpp:
```diff
@@ -28,7 +28,11 @@
 #if ENABLE(WK_WEB_EXTENSIONS)
 
+#if ENABLE(WPE_PLATFORM)
 #include "WPEToplevelPrivate.h"
```
Remove this, we shouldn't need it.

ItsJamie9494 committed:
[WPE] Fix build after 301229
https://bugs.webkit.org/show_bug.cgi?id=300684

Reviewed by Michael Catanzaro.

WPE didn't have the WebExtensions code enabled. Enabling it
reveals many bugs that assumed Gtk would always be present.
We can move a lot of code to GTK specific files and leave non-GTK parts
available for WPE.

This also disables the action-parsing test on WPE, since that
test creates a PNG image and WPE doesn't have any libraries that
support doing that.

* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/platform/graphics/Icon.cpp:
* Source/WebCore/platform/graphics/Icon.h:
* Source/WebCore/platform/graphics/glib/IconGLib.cpp: Renamed from Source/WebCore/platform/graphics/gtk/IconGtk.cpp.
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/Shared/API/APIData.h:
* Source/WebKit/Shared/Extensions/wpe/WebExtensionUtilitiesWPE.cpp:
(WebKit::availableScreenScales):
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/Extensions/glib/WebExtensionGLib.cpp: Copied from Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp.
* Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGTK.cpp: Added.
(WebKit::WebExtension::iconForPath):
* Source/WebKit/UIProcess/Extensions/wpe/WebExtensionWPE.cpp: Added.
(WebKit::WebExtension::iconForPath):
* Source/cmake/OptionsWPE.cmake:
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebExtension.cpp:
(beforeAll):
* Tools/TestWebKitAPI/glib/CMakeLists.txt:
* Tools/TestWebKitAPI/glib/PlatformGTK.cmake:
* Tools/TestWebKitAPI/glib/WebKitGLib/gtk/WebExtensionUtilities.cpp: Renamed from Tools/TestWebKitAPI/glib/WebKitGLib/WebExtensionUtilities.cpp.

Canonical link: https://commits.webkit.org/301906@main
